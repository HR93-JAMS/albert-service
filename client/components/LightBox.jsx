import React from 'react';

import Carousel from './Carousel.jsx';

import styles from '../styles/lightbox-style.css';
import arrowRight from '../icons/arrow-right.png';
import arrowLeft from '../icons/arrow-left.png';
import popupArrow from '../icons/popup-arrow.png';

class LightBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listOpen: true
    }
    this.togglePhotoList = this.togglePhotoList.bind(this);
  }

  onModalClick(e) {
    let backdrop = document.getElementById('backdrop');
    if (e.target === backdrop) {
      this.props.closeModal();
    }
  }

  togglePhotoList(keepOpen) {

    this.setState({
      listOpen: keepOpen || !this.state.listOpen
    })
  }

  render() {
    console.log('List open:' + this.state.listOpen)
    let { images, curImageIndex, closeModal, changeIndex } = this.props;
    let curImage = images[curImageIndex];
    return (
      <div id="backdrop" className={styles['lightbox-container']} onClick={(e) => this.onModalClick(e)}>
        <button id="prev" className={styles['nav-left']} onClick={() => changeIndex(--curImageIndex)}>
          <img className={styles['nav-icon']} src={arrowLeft} style={{left: 0}}/>
        </button>
        <img 
          key={curImageIndex} 
          src={curImage.src} 
          className={styles['display-image']} 
          align="middle" 
          onClick={() => changeIndex(++curImageIndex)}
        />
        <button id="next" className={styles['nav-right']} onClick={() => changeIndex(++curImageIndex)}>
          <img className={styles['nav-icon']} src={arrowRight} style={{right: 0}}/>
        </button>
        <div id="lightbox-bottom" className={styles['lightbox-bottom']} onMouseEnter={() => this.togglePhotoList(true)}>
          <div className={styles[`photo-list-${this.state.listOpen ? 'up' : 'down'}`]}>
            <div id="photo-list-header" className={styles['photo-list-header']}>
              <span id="caption">{`${curImageIndex+1}/${images.length}: ${curImage.caption}`}</span>
              <span style={{float: 'right'}}>
                <button id="photo-list-button" className={styles['photo-list-button']} onClick={() => this.togglePhotoList()}>
                  <span>{`${this.state.listOpen ? 'Hide' : 'Show'} photo list`}</span>
                  <span style={{ marginLeft: '1px'}}><img height="10px" src={popupArrow}/></span>
                </button>
              </span>
            </div>
            <Carousel 
              images={images} 
              changeIndex={changeIndex} 
              curImageIndex={curImageIndex} 
            />
          </div>
        </div>
      </div>
    );
  }
}

export default LightBox;