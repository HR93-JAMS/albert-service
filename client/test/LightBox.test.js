import React from 'react';
import LightBox from '../components/LightBox.jsx';
import sinon from 'sinon';
import { images3 as images } from './exampleData.json';

let fillProps = {
  images: images,
  curImageIndex: 0,
  changeIndex: () => {}
}

describe('Testing LightBox', () => {
  test('Should render LightBox', () =>{
    const LightBoxWrapper = shallow(<LightBox {...fillProps}/>);
    expect(LightBoxWrapper).toMatchSnapshot();
  }); 

  test('Should toggle listOpen state when togglePhotoList invoked', () => {
    const LightBoxWrapper = mount(<LightBox {...fillProps}/>);
    let listOpenBefore = LightBoxWrapper.state().listOpen;
    LightBoxWrapper.instance().togglePhotoList();
    LightBoxWrapper.update();
    let listOpenAfter = LightBoxWrapper.state().listOpen;
    expect(listOpenBefore === listOpenAfter).toBe(false);
  });

  test('Should leave listOpen state true when togglePhotoList invoked with true', () => {
    const LightBoxWrapper = mount(<LightBox {...fillProps}/>);
    let listOpenBefore = LightBoxWrapper.state().listOpen;
    LightBoxWrapper.instance().togglePhotoList(true);
    LightBoxWrapper.update();
    let listOpenAfter = LightBoxWrapper.state().listOpen;
    expect(listOpenBefore === listOpenAfter).toBe(true);
  });

  test('Should invoke changeIndex prop  when arrows and main image clicked', () => {
    sinon.spy(fillProps, 'changeIndex');
    const LightBoxWrapper = shallow(<LightBox {...fillProps}/>);
    LightBoxWrapper.find('#prev').simulate('click');
    LightBoxWrapper.find('#next').simulate('click');
    LightBoxWrapper.find('#display-image').simulate('click');
    expect(fillProps.changeIndex.calledThrice).toBe(true);
    fillProps.changeIndex.restore();
  })

});