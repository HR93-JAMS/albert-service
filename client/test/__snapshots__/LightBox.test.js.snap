// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LightBox Should render LightBox 1`] = `
<div
  id="lightbox-container"
>
  <button
    id="close-lightbox"
  >
    <svg
      fill="#ffffff"
      height="50px"
      version="1.1"
      viewBox="0 0 50 50"
      width="50px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#ffffff"
        id="surface1"
      >
        <path
          d="M 9.15625 6.3125 L 6.3125 9.15625 L 22.15625 25 L 6.21875 40.96875 L 9.03125 43.78125 L 25 27.84375 L 40.9375 43.78125 L 43.78125 40.9375 L 27.84375 25 L 43.6875 9.15625 L 40.84375 6.3125 L 25 22.15625 Z "
          fill="#ffffff"
        />
      </g>
    </svg>
  </button>
  <button
    id="prev"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "left": 0,
        }
      }
    >
      <svg
        height="80px"
        id="Capa_1"
        style={
          Object {
            "enableBackground": "new 0 0 377.406 377.406",
          }
        }
        version="1.1"
        viewBox="0 0 377.406 377.406"
        width="80px"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          transform="matrix(1 0 0 1 0 0)"
        >
          <g>
            <path
              className="active-path"
              d="M270.652,373.981c11.583,10.618,27.993-5.792,17.375-17.375C230.11,300.62,173.159,244.633,115.242,188.647   C173.159,132.661,230.11,76.675,288.027,20.688c11.583-10.618-5.792-27.993-17.375-17.375   C209.839,62.195,149.027,121.078,89.179,179.96c-4.826,4.826-4.826,12.549,0,17.375   C149.027,256.217,209.839,315.099,270.652,373.981z"
              data-old_color="#150B0B"
              data-original="#000000"
              fill="#FFFFFF"
            />
          </g>
        </g>
      </svg>
    </div>
  </button>
  <img
    align="middle"
    id="display-image"
    key="0"
    onClick={[Function]}
    src="http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg"
  />
  <button
    id="next"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "right": 0,
        }
      }
    >
      <svg
        height="80px"
        id="Capa_1"
        style={
          Object {
            "enableBackground": "new 0 0 377.406 377.406",
          }
        }
        version="1.1"
        viewBox="0 0 377.406 377.406"
        width="80px"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <g
          transform="matrix(-1 0 0 1 377.406 0)"
        >
          <g>
            <path
              className="active-path"
              d="M270.652,373.981c11.583,10.618,27.993-5.792,17.375-17.375C230.11,300.62,173.159,244.633,115.242,188.647   C173.159,132.661,230.11,76.675,288.027,20.688c11.583-10.618-5.792-27.993-17.375-17.375   C209.839,62.195,149.027,121.078,89.179,179.96c-4.826,4.826-4.826,12.549,0,17.375   C149.027,256.217,209.839,315.099,270.652,373.981z"
              data-old_color="#FBFAFA"
              data-original="#000000"
              fill="#FFFFFF"
            />
          </g>
        </g>
      </svg>
    </div>
  </button>
  <div
    id="lightbox-bottom"
    onMouseEnter={[Function]}
  >
    <div>
      <div
        id="photo-list-header"
      >
        <span
          id="caption"
        >
          1/6: I'm an example!
        </span>
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <button
            id="photo-list-button"
            onClick={[Function]}
          >
            <span>
              Hide photo list
            </span>
            <span
              style={
                Object {
                  "height": "10px",
                  "marginLeft": "1px",
                  "width": "10px",
                }
              }
            >
              <div>
                <img
                  height="10px"
                  src={Object {}}
                  width="10px"
                />
              </div>
            </span>
          </button>
        </span>
      </div>
      <Carousel
        changeIndex={[Function]}
        curImageIndex={0}
        images={
          Array [
            Object {
              "caption": "I'm an example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
            Object {
              "caption": "I'm another example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
            Object {
              "caption": "I'm an example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
            Object {
              "caption": "I'm another example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
            Object {
              "caption": "I'm an example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
            Object {
              "caption": "I'm another example!",
              "location_id": 3,
              "src": "http://assets1.ignimgs.com/2018/02/26/spongebob-blogroll-1519673777353_1280w.jpg",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
